keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -16,561 +15,245 @@ approvers:
-
-OBCs reference a storage class. The storage class references the external provisioner, defines a reclaim policy, and specifies object-store specific parameters, such as region, owner secret, bucket lifecycle policies, etc.
-**Note:** the original design uses storage classes but we are open to using a namespaced _BucketClass_ Custom Resource based on community input.
-
-### Moving Parts
-
-Before deploying an object bucket consuming app, the OB and OBC CRDs need to be created with RBAC rules granting _get_, _list_, _watch_, and _delete_ verbs.
-**Note:** OBCs do not need delete permission but OBs, secrets, and configmaps do.
-An object store needs to be created and referenced in the storage classes used by OBCs.
-The OBCs needed by the app must exist in the app's namespace.
-
-As is true for dynamic PV provisioning, a bucket provisioner pod needs to be running for each type of object-store existing in the Kubernetes cluster.
-For example, for AWS S3, the developer creates an OBC referencing a storage class which references the S3 store.
-This storage class needs to be created by the admin.
-The S3 provisioner pod watches for OBCs whose storage classes point to the AWS S3 provisioner, while ignoring all other OBCs.
-Likewise, the same cluster can also run the Rook-Ceph RGW provisioner, which also watches OBCs, only handling OBCs that reference storage classes which define ceph-rgw.
-
-#### Greenfield (new)
-
-An OBC for a new bucket is defined by its inclusion of a generated or static bucket name.
-In this case the referenced storage class omits a bucket name.
-A new bucket OBC triggers the correct provisioner to create a new bucket and the associated artifacts (credentials, policy, etc). 
-The library, in turn, creates a secret, configmap, and OB based on information returned by the provisioner.
-
-The secret contains the credentials needed to access the bucket and the configmap contains bucket endpoint information.
-**Note:** it was mentioned at the 2019 NA Kubecon storage face-to-face that to improve scaling we could decide to move the configmap info into the secret, thus reducing k8s resources required.
-The OB is an abstraction of the bucket and can save some state data needed by provisioners.
-The secret and configmap reside in the same namespace as the OBC, and the OB lives in the provisioner's namespace.
-The app pod consumes the secret and configmap and thus can reference the bucket.
-
-When a _greenfield_ OBC is deleted the associated provisioner is expected to delete the newly provisioned bucket and the related object store-specific artifacts.
-The library will delete the secret, configmap and OB.
+## Vocabulary
++  _Brownfield_ - buckets created outside the COSI system but granted access via COSI.
++ _Bucket_ - A container where objects reside, similar to a POSIX directory.
```

Comment:
How about:
"Bucket - A flat namespace where objects reside, similar to a POSIX directory.
"